files/nrpe/check_object.py

```diff
@@ -144,6 +168,82 @@ def execute(self):
     except:
       raise
 
+class SwiftPublicAvailability():
+  '''
+  Check Swift API call length by listing contents of a public bucket
+  '''
+  options = dict()
+
+  def __init__(self, options):
+
+    self.creds = OSCredentials(options)
+
+    try:
+      if options.swift_host is not None:
+        self.swift_host = options.swift_host
+      else:
+        self.swift_host = os.environ['swift_host']
+    except KeyError: 
+      raise CredentialsMissingException(key='swift_host')
```

Comment:
This now ties into the change about how test parameters vs. credential parameters are passed on to the test class (thanks for modifying that part!). So same thing, a test parameter missing should not trigger something that in it's label has only the word "Credential".

Would you buy

1) 

Just giving it a bit wider definition e.g. `CredentialsOrParametersMissingException`

or

2) separating the two (different classes for missing creds vs. missing other parameters?)

Reply:
Done.